### PR TITLE
Add libstempo recipe

### PR DESCRIPTION
**Pull request by mattpitkin**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mattpitkin** [comment]
@conda-forge/help-python - could we initiate a review for this package?

@vallis could you add a comment on this PR that you are happy to be listed as a maintainer of the package?

File: recipes/libstempo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  number: 0
+  skip: true  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - astropy
```
Comment (dopplershift) [inline recipes/libstempo/meta.yaml:22]: I know `scipy` doesn't have a compiled API you'd be linking against, and I don't think `astropy` does either, so you *shouldn't* need them in `host`? Does that also apply to `tempo2`?

Comment (mattpitkin) [reply recipes/libstempo/meta.yaml]: Thanks. I've removed scipy and astropy from host. Tempo2 is compiled and linked to the cython compiled library within libstempo, so I think may be needed.

Comment (dopplershift) [reply recipes/libstempo/meta.yaml]: 👍

**Review by dopplershift — APPROVED** [review]
Looks good pending word from @vallis.

OSX failure is spurious (HTTP timeout, not package build failure)

**Comment by mattpitkin** [comment]
@vallis are you happy to be listed as a maintainer? If not, I can remove you and I'm happy to be the sole maintainer for the conda-forge package.

**Comment by dopplershift** [comment]
@mattpitkin If you want to get this in, you can remove @vallis from the recipe here, and add them (upon their approval) in a PR on the feedstock later.

**Comment by mattpitkin** [comment]
@dopplershift I've edited the meta.yaml file so that only I'm listed as a maintainer for the moment.

**Pull request merged**